Sources/Sentry/SentryUIPerformanceTracker.m

```diff
@@ -0,0 +1,258 @@
+    [SentryUIPerformanceTracker swizzleViewControllerInits];
+#else
+    [SentryLog logWithMessage:@"NO UIKit -> [SentryUIPerformanceTracker "
+                              @"start] does nothing."
+                     andLevel:kSentryLevelDebug];
+#endif
+}
+
+/**
+ * Swizzle the some init methods of the view controller,
+ * so we can swizzle user view controller subclass on demand.
+ */
++ (void)swizzleViewControllerInits
+{
+#if SENTRY_HAS_UIKIT
+    // SentrySwizzleInstanceMethod declaration shadows a local variable. The swizzling is working
+    // fine and we accept this warning.
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wshadow"
+
+    static const void *swizzleViewControllerInitWithCoder = &swizzleViewControllerInitWithCoder;
+
+    SEL coderSelector = NSSelectorFromString(@"initWithCoder:");
+    SentrySwizzleInstanceMethod(UIViewController.class, coderSelector, SentrySWReturnType(id),
+        SentrySWArguments(NSCoder * coder), SentrySWReplacement({
+            [SentryUIPerformanceTracker swizzleViewControllerSubClass:[self class]];
+            return SentrySWCallOriginal(coder);
+        }),
+        SentrySwizzleModeOncePerClassAndSuperclasses, swizzleViewControllerInitWithCoder);
+
+    static const void *swizzleViewControllerInitWithNib = &swizzleViewControllerInitWithNib;
+    SEL nibSelector = NSSelectorFromString(@"initWithNibName:bundle:");
+    SentrySwizzleInstanceMethod(UIViewController.class, nibSelector, SentrySWReturnType(id),
+        SentrySWArguments(NSString * nibName, NSBundle * bundle), SentrySWReplacement({
+            [SentryUIPerformanceTracker swizzleViewControllerSubClass:[self class]];
+            return SentrySWCallOriginal(nibName, bundle);
+        }),
+        SentrySwizzleModeOncePerClassAndSuperclasses, swizzleViewControllerInitWithNib);
+#    pragma clang diagnostic pop
+#endif
+}
+
++ (void)swizzleViewControllerSubClass:(Class)class
+{
+#if SENTRY_HAS_UIKIT
+    // Swizzling only classes from the user app module to avoid track every UIKit view controller
+    // interaction.
+    static const char *appImage = nil;
+    if (appImage == nil) {
+        if ([UIApplication respondsToSelector:@selector(sharedApplication)]) {
+            UIApplication *app = [UIApplication performSelector:@selector(sharedApplication)];
+            appImage = class_getImageName(app.delegate.class);
+        }
+    }
+    if (appImage == nil || strcmp(appImage, class_getImageName(class)) != 0)
+        return;
+
+    // This are the five main functions related to UI creation in a view controller.
+    // We are swizzling it to track anything that happens inside one of this functions.
+    [SentryUIPerformanceTracker swizzleViewLayoutSubViews:class];
+    [SentryUIPerformanceTracker swizzleLoadView:class];
+    [SentryUIPerformanceTracker swizzleViewDidLoad:class];
+    [SentryUIPerformanceTracker swizzleViewWillAppear:class];
+    [SentryUIPerformanceTracker swizzleViewDidAppear:class];
```

Comment:
`h`: It would be nice to create spans for the different phases of the view controller. We can determine how load each of these lifecycle methods took, so we could add spans for each of them.

Reply:
I like the idea!!

![image](https://user-images.githubusercontent.com/28265868/118069419-88f3e780-b37a-11eb-92dd-3f583a2cbbec.png)

Reply:
I don't think so, I never see anybody overrating loadViewIfNeeded. And all it does is call loadView if the view is nil. 
The right function to override would be loadView.